src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperScopeManager.cs

```diff
@@ -32,17 +33,35 @@ public TagHelperScopeManager()
         /// <param name="startWritingScope">A delegate used to start a writing scope in a Razor page.</param>
         /// <param name="endWritingScope">A delegate used to end a writing scope in a Razor page.</param>
         /// <returns>A <see cref="TagHelperExecutionContext"/> to use.</returns>
-        public TagHelperExecutionContext Begin([NotNull] string tagName,
-                                               [NotNull] string uniqueId,
-                                               [NotNull] Func<Task> executeChildContentAsync,
-                                               [NotNull] Action startWritingScope,
-                                               [NotNull] Func<TextWriter> endWritingScope)
+        public TagHelperExecutionContext Begin(
+            [NotNull] string tagName,
+            [NotNull] string uniqueId,
+            [NotNull] Func<Task> executeChildContentAsync,
+            [NotNull] Action startWritingScope,
+            [NotNull] Func<TextWriter> endWritingScope)
         {
-            var executionContext = new TagHelperExecutionContext(tagName,
-                                                                 uniqueId,
-                                                                 executeChildContentAsync,
-                                                                 startWritingScope,
-                                                                 endWritingScope);
+            IDictionary<object, object> items;
+            var parentExecutionContext = _executionScopes.Count > 0 ? _executionScopes.Peek() : null;
+
+            // If we're not wrapped by another TagHelper, then there will not be a parentExecutionContext.
+            if (parentExecutionContext != null)
```

Comment:
nit: this could be `if (_executionScopes.Count != 0)` (or `> 0` as you prefer).  then you wouldn't need the ternary above